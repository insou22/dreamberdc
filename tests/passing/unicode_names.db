const const firstAlphabetLetter = "A"!
var const 👍 = true!
var var 1️⃣ = 1! 