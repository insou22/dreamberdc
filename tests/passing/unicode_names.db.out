Program {
    items: [
        Statement(
            Assignment(
                AssignmentStatement {
                    decl_type: ConstConst,
                    lhs: Ident(
                        Ident {
                            name: "firstAlphabetLetter",
                        },
                    ),
                    rhs: StringLit(
                        StringLit {
                            contents: "A",
                        },
                    ),
                    termination: Importance(
                        1,
                    ),
                },
            ),
        ),
        Statement(
            Assignment(
                AssignmentStatement {
                    decl_type: VarConst,
                    lhs: Ident(
                        Ident {
                            name: "üëç",
                        },
                    ),
                    rhs: BoolLit(
                        True,
                    ),
                    termination: Importance(
                        1,
                    ),
                },
            ),
        ),
        Statement(
            Assignment(
                AssignmentStatement {
                    decl_type: VarVar,
                    lhs: Ident(
                        Ident {
                            name: "1\u{fe0f}\u{20e3}",
                        },
                    ),
                    rhs: NumericLit(
                        NumericLit {
                            digits: "1",
                        },
                    ),
                    termination: Importance(
                        1,
                    ),
                },
            ),
        ),
    ],
}